File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.cpp
class Solution {
public:
    int specialArray(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int n=nums.size();
        int pre=0;
        for(int i=0;i<n;i++)
        {
            int val =n-i;
            if(val<=nums[i] && val>pre)return val;
            pre=nums[i];
        }
        return -1;
    }
};

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        for i in range(len(nums),0,-1):
            count=0
            for j in range(len(nums)):
                if nums[j]>=i:
                    count+=1
            if count==i:
                return i
        return -1